### ISSUE TITLE
Code for training llama pro?

### ISSUE THREAD

**Issue by yhyu13** [TencentARC/LLaMA-Pro#2]
Hi,

If I get it correctly, you have used code from https://github.com/allenai/open-instruct as base.

Would you release the full code of reproducing llama2 pro 8B?

Thanks!

**Comment by hills-code** [comment]
Yes, of course. I will organize the code recently. Thanks for your interest.

**Comment by raghavgarg97** [comment]
Hey @hills-code ,could you also add code for converting a model by adding identity blocks for training ?
I am excited to use similar techniques for other open-source models !!
Thanks!:)

**Comment by hills-code** [comment]
> Hey @hills-code ,could you also add code for converting a model by adding identity blocks for training ? I am excited to use similar techniques for other open-source models like Qwen!! Thanks!:)

I have added the block expansion script under the folder scripts. You can check it for reference. Hope it will be helpful!

**Comment by raghavgarg97** [comment]
i will check that out,Thanks!

**Comment by raghavgarg97** [comment]
@hills-code  i was able to get it to work! Btw you only train the added layers and not even the lm head?

**Comment by yhyu13** [comment]
@raghavgarg97 No, I don't think you can skip the "post-training" for extra blocks on some new corpus (e.g. bigcode, as mentioned in the paper) before applying tuning

**Comment by Abolfazl-kr** [comment]
does anybody know how to continue pre-training? @yhyu13 @raghavgarg97 @yxgeee @hills-code

**Comment by yhyu13** [comment]
nah，code has not been released yet
